fsthttp/request.go

```diff
@@ -209,8 +211,30 @@ func newClientRequest() (*Request, error) {
 	}, nil
 }
 
+// SetBody sets the [Request]'s body to the provided [io.Reader]. Prefer
+// using this method over setting the Body field directly, as it enables
+// optimizations in the runtime.
+//
+// If an unread body from an incoming client request is set on an
+// outgoing upstream request, the body will be efficiently streamed from
+// the incoming request.  It is also possible to set the unread body of
+// a received response to the body of a request, with the same results.
+//
+// If the body is set from an in-memory reader such as [bytes.Buffer],
+// [bytes.Reader], or [strings.Reader], the runtime will send the
+// request with a Content-Length header instead of Transfer-Encoding:
+// chunked.
+func (req *Request) SetBody(body io.Reader) {
```

Comment:
A bit of a learning question here, but is it possible to have the case of and incoming request with a content-length brought to this body assignment as well?

Reply:
Realized this isn't super clear, but I think `Clone` already handles this by copying the header if present.

Reply:
Yes, but the runtime will remove `Content-Length` and `Transfer-Encoding` before sending the request unless `ManualFramingMode` is true.